src/modals/Modal.tsx

```diff
@@ -106,8 +112,44 @@ export class Modal extends React.PureComponent<ModalProps> {
         this.cleanup();
     }
 
+    getTitleIconByType(type: ModalType): React.ReactNode {
+        switch (type) {
+            case ModalType.INFO:
+                return <Icon aria-hidden={true} shape="info-standard" />;
+            case ModalType.WARNING:
+                return <Icon aria-hidden={true} shape="warning-standard" />;
+            case ModalType.DANGER:
+                return <Icon aria-hidden={true} shape="error-standard" />;
+            default:
+                return;
+        }
+    }
+
+    /*Function to build title for modal
+      overwrites type icons if custom icon is provided, doesn't renders any icon if showIcon is set to false
+    */
+    buildTitle(): React.ReactElement {
+        const {type, customIcon, title} = this.props;
+        const {showIcon} = this.state;
+        let titleIcon: React.ReactNode | undefined = customIcon
+            ? customIcon
+            : type && type !== ModalType.DEFAULT
+            ? this.getTitleIconByType(type)
+            : undefined;
```

Comment:
```suggestion
        const titleIcon: React.ReactNode | undefined = customIcon
            ? customIcon :        
             this.getTitleIconByType(type)
           
```

Reply:
The ModalType defualt need not to have any Icons hence this condition is added, if modal type is set as default no pre defined icon is set

Reply:
@tejasjc-ecs  My suggested code block will also work with Default Type. can u give a try ?

Reply:
Yes, the suggested code worked as expected as default case returning undefined already, thanks! Done this change